tests/e2e/controls.py

```diff
@@ -2717,3 +2747,252 @@ def expect_icon(self, value: PatternOrStr, *, timeout: Timeout = None) -> None:
 
     def expect_open(self, is_open: bool, *, timeout: Timeout = None) -> None:
         _expect_class_value(self.loc_body, "show", is_open, timeout=timeout)
+
+    # user sends value of Open: true | false
+    def set(self, open: bool, *, timeout: Timeout = None) -> None:
+        if self.loc_body_visible.count() != int(open):
+            self.toggle(timeout=timeout)
+
+    def toggle(self, *, timeout: Timeout = None) -> None:
+        self.loc_header.click(timeout=timeout)
+
+
+class _OverlayBase(_InputBase):
+    def __init__(
+        self,
+        page: Page,
+        *,
+        id: str,
+        loc: InitLocator,
+        overlay_name: str,
+        overlay_selector: str,
+    ) -> None:
+        super().__init__(page, id=id, loc=loc)
+        self._overlay_name = overlay_name
+        self._overlay_selector = overlay_selector
+
+    def _get_overlay_id(self) -> str | None:
+        loc_el = self.loc.locator(
+            f" > :last-child[data-bs-toggle='{self._overlay_name}']"
+        )
+        return loc_el.get_attribute("aria-describedby")
```

Comment:
Let's leverage the `@property` function like in the nav item below and have `self.loc_overlay_body` be a getter method. This will keep the API consistent if we can ever make it a single locator.